syft/execution/action.py

```diff
@@ -10,4 +20,258 @@ class Action(ABC):
     tensors and add them together is an Action. Sending an object from one worker to another is
     also an Action."""
 
-    pass
+    def __init__(
+        self, name: str, target, args_: tuple, kwargs_: dict, return_ids: tuple, return_value=False
+    ):
+        """Initialize an action
+
+        Args:
+            name (String): The name of the method to be invoked (e.g. "__add__")
+            target (Tensor): The object to invoke the method on
+            args_ (Tuple): The arguments to the method call
+            kwargs_ (Dictionary): The keyword arguments to the method call
+            return_ids (Tuple): primarily for our async infrastructure (Plan, Protocol, etc.), the id of
+                action results are set by the client. This allows the client to be able to predict where
+                the results will be ahead of time. Importantly, this allows the client to pre-initalize the
+                pointers to the future data, regardless of whether the action has yet executed. It also
+                reduces the size of the response from the action (which is very often empty).
+            return_value (boolean): return the result or not. If true, the result is directly returned,
```

Comment:
I know this has nothing to do with this PR but could `return_value ` be inferred from the method name?

Reply:
Maybe? Not sure, but might be worth a look. Create an issue?

Reply:
Anyway, it's a detail